src/DI/WebpackExtension.php

```diff
@@ -110,4 +114,44 @@ public function beforeCompile()
 		}
 	}
 
+
+	private function setupAssetResolver(array $config): ServiceDefinition
+	{
+		$builder = $this->getContainerBuilder();
+
+		$assetResolver = $builder->addDefinition($this->prefix('assetResolver'))
+			->setClass(AssetResolver\AssetResolverInterface::class);
+
+		if ($config['manifest']['name'] !== NULL) {
+			Validators::assertField($config['manifest'], 'name', 'string:1..');
+			$loader = $builder->addDefinition($this->prefix('manifestLoader'))
+				->setClass(ManifestLoader::class);
+
+			$args = [
+				$config['manifest']['name'],
+				$loader,
+			];
+
+			if ($config['manifest']['optimize']) {
+				$devServerInstance = new DevServer($config['devServer']['enabled'], $config['devServer']['url'], new Client());
+				$directoryProviderInstance = new BuildDirectoryProvider($config['build']['directory'], $devServerInstance);
```

Comment:
AssetLocator is a brilliant idea, I had thought of it before but hadn't come up with a right way to approach it, so thanks for pointing out :) implemented in 28c6fc1 and marked `BuildDirectoryProvider` and `PublicPathProvider` as internal so that `AssetLocator` can become the official public API.

The two providers, however, should imo stay as they are. They are used in different contexts and thus I consider them to be two different responsibilities